### PR TITLE
[OpenVINO] Implement flash_attn and bias support

### LINKED ISSUE #35585: [Keras 3 OpenVINO backend] Implement flash_attn and bias support
### Request Description

This issue tracks the implementation of flash_attention and bias support in the Keras OpenVINO backend. This will help increase the coverage of functional parity for the OpenVINO backend.

I will submit a PR implementing this and enabling the corresponding tests.

Ref: https://github.com/openvinotoolkit/openvino/issues/34017

### Feature Use Case

_No response_

### Issue submission checklist

- [x] The feature request or improvement must be related to OpenVINO

### PR DESCRIPTION
**Pull request by goyaladitya05**
## Description

bias is now supported by adding it to the attn mask before passing to scaled_dot_product_attention (which treats attention_mask as additive). Boolean masks are converted to float before addition. 

Removed the flash_attention guard since scaled_dot_product_attention is already a fused kernel that covers it.

Closes: openvinotoolkit/openvino/issues/35585

## Contributor Agreement
**Please review our
[AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy)
and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

**Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

File: keras/src/backend/openvino/nn.py [hunk 1]
```diff
@@ -1265,6 +1255,22 @@ def dot_product_attention(
...
     mask = get_ov_output(mask) if mask is not None else None
+    if bias is not None:
+        bias = get_ov_output(bias)
+        if bias.get_element_type() != query.get_element_type():
+            bias = ov_opset.convert(bias, query.get_element_type()).output(0)
+        if mask is not None:
+            if mask.get_element_type() == Type.boolean:
+                large_neg = ov_opset.constant(
+                    np.finfo(np.float32).min, query.get_element_type()
+                ).output(0)
```
[the remark this answers is not in the document]
Comment (goyaladitya05) [reply keras/src/backend/openvino/nn.py:1272]: good catch, addressed.

**Review by hertschuh — APPROVED** [review]
Great, thanks!
